hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/SnapshotChainManager.java

```diff
@@ -105,7 +125,26 @@ private void addSnapshotPath(String snapshotPath,
           snapshotID));
     }
 
+    if (prevPathID == null && snapshotChainByPath.containsKey(snapshotPath) &&
+        !snapshotChainByPath.get(snapshotPath).isEmpty()) {
+      throw new IllegalStateException(String.format(
+          "Snapshot chain corruption. Error while adding snapshot with " +
+              "snapshotId %s with as the first snapshot in snapshot path:" +
+              " %s which already %d snapshots.", snapshotID, snapshotPath,
+          snapshotChainByPath.get(snapshotPath).size()));
+    }
+
     if (prevPathID != null && snapshotChainByPath.containsKey(snapshotPath)) {
+      if (snapshotChainByPath.get(snapshotPath).get(prevPathID)
+          .getNextSnapshotId() != null) {
+        throw new IOException(String.format("Snapshot chain corruption. " +
```

Comment:
Please use `IllegalStateException`. Also change it for line number 100 and 122 and remove `IOException` from the method signature as well.

Reply:
done